### PR TITLE
early check for bad package name

### PR DESCRIPTION
**Pull request by gilesknap**
Check package name at start of 'new' command to avoid creating git repo before failing.

### CODE REVIEW

**Comment by gilesknap** [comment]
ooops - did not adhere to protocol 

I guess I need to enable git hooks

File: src/dls_python3_skeleton/__main__.py [hunk 1]
```diff
@@ -125,6 +126,11 @@ def replace_text(text: str) -> str:
 
 def new(args):
     path: Path = args.path
+
+    package = path.name
```
Comment (thomascobb) [inline src/dls_python3_skeleton/__main__.py:130]: You need this
```python
package = args.package or path.name
```
But this is duplication.

I suggest putting the following in a function and calling it from both `new` and `existing`:
```python
    package = override_package or repo
    valid = VALID_PKG.match(package)
    assert valid, f"'{package}' is not a valid python package name"
```

**Review by thomascobb — CHANGES_REQUESTED** [review]

File: src/dls_python3_skeleton/__main__.py (hunk 1 above)
Comment (gilesknap) [reply src/dls_python3_skeleton/__main__.py:130]: Yes I worked that out - I got a little confused about which args went to which command. 

Do you not want to include typer as part of skeleton? That argparse looks way old fashioned. As the skeleton provider you get to be opinionated about such things.

Comment (thomascobb) [reply src/dls_python3_skeleton/__main__.py:130]: I specifically chose argparse so it added no external dependencies, and I think that's probably the right approach here...

Comment (gilesknap) [reply src/dls_python3_skeleton/__main__.py]: I guess it would be the only non-standard library in the runtime dependencies though.

Comment (gilesknap) [reply src/dls_python3_skeleton/__main__.py]: I made it DRYer

Comment (thomascobb) [reply src/dls_python3_skeleton/__main__.py]: Skeleton likes DRY...

**Comment by gilesknap** [comment]
Yes, that would be more robust. Sorry for delays, I'm not getting any notifications from github (and they are not going to junk either)

**Review by thomascobb — APPROVED** [review]

**Pull request merged**